pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -316,7 +326,7 @@ func (p *Provider) InstallCustomProviderComponents(ctx context.Context, kubeconf
 	return nil
 }
 
-func (p *Provider) InstallTinkerbellStack(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
+func (p *Provider) InstallTinkerbellStack(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, bootstrap bool) error {
```

Comment:
Please can we rethink this. Passing bool flags to toggle behavior creates hard to follow branch logic that becomes increasingly complex with time.

Reply:
Yeah, I am planning to rework this as part of my next PR which will change some of this logic. Are you okay with this for now?